dash/include/dash/Atomic.h

```diff
@@ -34,58 +39,56 @@ class Atomic
    * Default constructor.
    * Sets underlying global pointer to \c DART_GPTR_NULL.
    */
-  Atomic()
-  : _gptr(DART_GPTR_NULL),
-    _team(nullptr)
-  { }
+  AtomicAddress() = default;
 
   /**
-   * Constructor, creates a new instance of \c dash::Atomic from a DART
+   * Constructor, creates a new instance of \c dash::AtomicAddress from a DART
    * pointer.
    */
-  Atomic(
-    dart_gptr_t  gptr,
+  AtomicAddress(
+    const dart_gptr_t & gptr,
     dash::Team & team = dash::Team::All())
   : _gptr(gptr),
-    _team(&team)
+    _dart_teamid(team.dart_id())
   { }
 
   /**
-   * Constructor, creates a new instance of \c dash::Atomic from any object
+   * Constructor, creates a new instance of \c dash::AtomicAddress from any object
    * that is convertible to \c dart_gptr_t.
    */
   template<class GlobalType>
-  Atomic(
+  AtomicAddress(
     const GlobalType & global,
     dash::Team       & team   = dash::Team::All())
-  : Atomic(global.dart_gptr(), team)
+  : AtomicAddress(global.dart_gptr(), team)
   { }
 
-  Atomic(const self_t & other)           = default;
+  AtomicAddress(const self_t & other)    = default;
 
   self_t & operator=(const self_t & rhs) = default;
 
   /**
    * Set the value of the shared atomic variable.
    */
-  void set(ValueType val)
+  void set(ValueType value)
   {
-    DASH_LOG_DEBUG_VAR("Atomic.set()", val);
-    DASH_LOG_TRACE_VAR("Atomic.set",   _gptr);
-    *(reference(_gptr)) = val;
-    DASH_LOG_DEBUG("Atomic.set >");
+    DASH_LOG_DEBUG_VAR("AtomicAddress.set()", value);
+    DASH_LOG_TRACE_VAR("AtomicAddress.set",   _gptr);
+    *(pointer(_gptr)) = value;
+    dart_flush_all(_gptr);
```

Comment:
Two things: 

1. Is the flush really required? The assignment should use `dart_put_blocking`, which issues a flush at the end. 
2. More importantly: This operation is not atomic and might thus conflict with other atomic operations executed on this memory. I suggest using `dart_accumulate`, which is guaranteed to be atomic (implemented in terms of `MPI_Accumulate`.